### ISSUE TITLE
Projects: add better default sorting to version listing

### ISSUE THREAD

**Issue by agjohnson** [readthedocs/readthedocs.org#11510]
The project version listing currently defaults to `recently built` sorting, which helps some, but does hide important versions like `latest`/`stable`.

I had originally tried to add a _sort by relevance_ sorting method, but felt I was hacking up the queryset too much. But it seems really doable and would be better default UX.

I brought this up in this issue as a way to surface latest/stable higher in the version list. But either way I think it would be a better default sorting, regardless of what we do with a button.

- https://github.com/readthedocs/ext-theme/issues/278

And to clarify, this will not alter any other filter querysets.

- [ ] Add a new project version sorting method, _sort by relevance_. This would be sort by recently built, plus latest/stable first in the query.
- [ ] Use this as the default sorting method on the version listing view.

**Comment by humitos** [comment]
> This would be sort by recently built, plus latest/stable first in the query.

I would probably say "Default version", Latest and Stable. That way, it's easier to find the _actual_ default version first.

Also, I would suggest to distinct visually these 3 from the rest of the items. Maybe a different block?

**Comment by agjohnson** [comment]
There are some options here, yeah.

I wouldn't move these to separate block or style the rows different. I'd like to retain the same UI listing in a consistent manner across all UIs.

But we could color the version's buttons primary to make those rows stand out, and there is already a pattern for using labels to describe special version states:

![image](https://github.com/user-attachments/assets/46833b9d-8b9c-4c45-87dc-4dd08243b11d)

I'd like to revisit that pattern and tune up what we're using those labels for.

**Comment by humitos** [comment]
Actually, using a different color for the "Default" label would be a great first step to me 👍🏼

**Comment by agjohnson** [comment]
Yeah, that's what I was thinking too. Matching colors on the buttons and labels, should be a fair amount of visual distinction already. I'd still like to see them at the top of the list to further help identify them as important versions, but both in combination sounds like much better UX.

We can do the color changes separate, that's an easier change.
